fendermint/vm/topdown/proof-service/src/config.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! Configuration for the proof generator service
+
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+/// Configuration for the proof generator service
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProofServiceConfig {
+    /// Enable/disable the service
+    pub enabled: bool,
+
+    /// Polling interval for checking parent chain
+    #[serde(with = "humantime_serde")]
+    pub polling_interval: Duration,
+
+    /// How many instances ahead to generate proofs (lookahead window)
+    pub lookahead_instances: u64,
+
+    /// How many old instances to retain after commitment
+    pub retention_instances: u64,
+
+    /// Lotus/parent RPC endpoint URL
+    pub parent_rpc_url: String,
+
+    /// Parent subnet ID (e.g., "/r314159" for calibration)
+    pub parent_subnet_id: String,
+
+    /// F3 network name (e.g., "calibrationnet", "mainnet")
+    pub f3_network_name: String,
+
+    /// Optional: Additional RPC URLs for failover (not yet implemented - future enhancement)
+    #[serde(default)]
+    pub fallback_rpc_urls: Vec<String>,
+
+    /// Maximum cache size in bytes (0 = unlimited)
+    #[serde(default)]
+    pub max_cache_size_bytes: usize,
+
+    /// Gateway actor on parent chain (for proof generation).
+    ///
+    /// Can be either:
+    /// - Actor ID: 176609
+    /// - Ethereum address: 0xE4c61299c16323C4B58376b60A77F68Aa59afC8b (will be resolved to actor ID)
+    ///
+    /// Will be configured from subnet genesis info.
+    #[serde(default)]
+    pub gateway_actor_id: Option<u64>,
+
+    /// Gateway ethereum address (alternative to gateway_actor_id).
+    ///
+    /// If provided, will be resolved to actor ID on service startup.
+    #[serde(default)]
+    pub gateway_eth_address: Option<String>,
+
+    /// Subnet ID (for event filtering)
+    /// Will be derived from genesis
+    #[serde(default)]
+    pub subnet_id: Option<String>,
+
+    /// Maximum epoch lag before considering a certificate too old to generate proofs for.
+    /// If a certificate's highest epoch is more than this many epochs behind the current
+    /// parent chain epoch, it will be skipped.
+    /// Default: 100 epochs (~50 minutes on Filecoin mainnet)
+    #[serde(default = "default_max_epoch_lag")]
+    pub max_epoch_lag: u64,
+
+    /// Maximum lookback window that the parent RPC supports.
+    /// Most Lotus nodes have a limited lookback window (e.g., 2000 epochs).
+    /// If we're further behind than this, we can't generate proofs.
+    /// Default: 2000 epochs
+    #[serde(default = "default_rpc_lookback_limit")]
+    pub rpc_lookback_limit: u64,
```

Comment:
These parameters are not used in this PR. Do we really want them?

Reply:
I will remove it. It is from previous version..